src/Analyzers/Core/Analyzers/RemoveUnnecessarySuppressions/AbstractRemoveUnnecessaryPragmaSuppressionsDiagnosticAnalyzer.cs

```diff
@@ -99,7 +99,12 @@ protected sealed override void InitializeWorker(AnalysisContext context)
             // Bail out if analyzer is suppressed on this file or project.
             // NOTE: Normally, we would not require this check in the analyzer as the analyzer driver has this optimization.
             // However, this is a special analyzer that is directly invoked by the analysis host (IDE), so we do this check here.
-            if (tree.DiagnosticOptions.TryGetValue(IDEDiagnosticIds.RemoveUnnecessarySuppressionDiagnosticId, out var severity) ||
+            ReportDiagnostic severity;
+            if (
+#if !CODE_STYLE
+                compilationWithAnalyzers.Compilation.Options.SyntaxTreeOptionsProvider != null &&
+                compilationWithAnalyzers.Compilation.Options.SyntaxTreeOptionsProvider.TryGetDiagnosticValue(tree, IDEDiagnosticIds.RemoveUnnecessarySuppressionDiagnosticId, out severity) ||
+#endif
```

Comment:
@mavasani What compat requirements do we need with this analyzer? Do I need to also check the now deprecated property for down-level compilers?

Reply:
No, this is an IDE-only analyzer that was recently added. It is currently a no-op in CodeStyle NuGet package, so we should be fine assuming this ships just with VS and change it as appropriate in this PR.

Reply:
@mavasani Alright then this is probably fine, unless you want to simply put the #if CODE_STYLE somewhere else to make it clear this entire analyzer is IDE only.